### ISSUE TITLE
dataset to test the model, and evaluate the model. 

### ISSUE THREAD

**Issue by KeenNest** [vb000/NeuriCam#11]
To run this command  , 
python evaluate.py --lr_dir=<path to LR dir> --key_dir=<path to Key dir> --target_dir=<ground-truth HR dir> --model_dir=experiments/bix4_keyvsrc_attn --restore_file=pretrained --file_fmt=<file format eg., "%08d.png">

i want these paths right -path of LR,Key and ground-truth .
can you please share the dataset link from where i download .

**Comment by KeenNest** [comment]
After this :- python evaluate.py --lr_dir= --key_dir= --target_dir= --model_dir=experiments/bix4_keyvsrc_attn --restore_file=pretrained --file_fmt=<file format eg., "%08d.png">  command I am getting error like this,

RuntimeError: Error(s) in loading state_dict for Net:
        Unexpected key(s) in state_dict: "model.basicvsr_pp.deform_align.backward_1.weight", "model.basicvsr_pp.deform_align.backward_1.bias", "model.basicvsr_pp.deform_align.backward_1.conv_offset.0.weight", "model.basicvsr_pp.deform_align.backward_1.conv_offset.0.bias", "model.basicvsr_pp.deform_align.backward_1.conv_offset.2.weight", "model.basicvsr_pp.deform_align.backward_1.conv_offset.2.bias", "model.basicvsr_pp.deform_align.backward_1.conv_offset.4.weight", "model.basicvsr_pp.deform_align.backward_1.conv_offset.4.bias", "model.basicvsr_pp.deform_align.backward_1.conv_offset.6.weight", "model.basicvsr_pp.deform_align.backward_1.conv_offset.6.bias", "model.basicvsr_pp.deform_align.forward_1.weight", "model.basicvsr_pp.deform_align.forward_1.bias", "model.basicvsr_pp.deform_align.forward_1.conv_offset.0.weight", "model.basicvsr_pp.deform_align.forward_1.conv_offset.0.bias", "model.basicvsr_pp.deform_align.forward_1.conv_offset.2.weight", "model.basicvsr_pp.deform_align.forward_1.conv_offset.2.bias", "model.basicvsr_pp.deform_align.forward_1.conv_offset.4.weight", "model.basicvsr_pp.deform_align.forward_1.conv_offset.4.bias", "model.basicvsr_pp.deform_align.forward_1.conv_offset.6.weight", "model.basicvsr_pp.deform_align.forward_1.conv_offset.6.bias", "model.basicvsr_pp.deform_align.backward_2.weight", "model.basicvsr_pp.deform_align.backward_2.bias", "model.basicvsr_pp.deform_align.backward_2.conv_offset.0.weight", "model.basicvsr_pp.deform_align.backward_2.conv_offset.0.bias", "model.basicvsr_pp.deform_align.backward_2.conv_offset.2.weight", "model.basicvsr_pp.deform_align.backward_2.conv_offset.2.bias", "model.basicvsr_pp.deform_align.backward_2.conv_offset.4.weight", "model.basicvsr_pp.deform_align.backward_2.conv_offset.4.bias", "model.basicvsr_pp.deform_align.backward_2.conv_offset.6.weight", "model.basicvsr_pp.deform_align.backward_2.conv_offset.6.bias", "model.basicvsr_pp.deform_align.forward_2.weight", "model.basicvsr_pp.deform_align.forward_2.bias", "model.basicvsr_pp.deform_align.forward_2.conv_offset.0.weight", "model.basicvsr_pp.deform_align.forward_2.conv_offset.0.bias", "model.basicvsr_pp.deform_align.forward_2.conv_offset.2.weight", "model.basicvsr_pp.deform_align.forward_2.conv_offset.2.bias", "model.basicvsr_pp.deform_align.forward_2.conv_offset.4.weight", "model.basicvsr_pp.deform_align.forward_2.conv_offset.4.bias", "model.basicvsr_pp.deform_align.forward_2.conv_offset.6.weight", "model.basicvsr_pp.deform_align.forward_2.conv_offset.6.bias".

**Comment by KeenNest** [comment]
why i am getting this error while running evaluate.py 
...,
           [0.6902, 0.7098, 0.7176,  ..., 0.2078, 0.2078, 0.2157],
           [0.6941, 0.7098, 0.7176,  ..., 0.1804, 0.1451, 0.1608],
           [0.6784, 0.7294, 0.7412,  ..., 0.1882, 0.1961, 0.1843]]]]]), 'key_frame_int': tensor([15])} ('office-video',)
Illegal instruction (core dumped)

**Comment by Justarrrrr** [comment]
hello, I meet the same demand, can you please share the dataset link from where i download

**Comment by KeenNest** [comment]
@Justarrrrr i created my own dataset ,
if you're getting illegal instruction (core dumped ) then reduce the size of frame to 160 * 120

Thanks

**Comment by Justarrrrr** [comment]
hello,I have other problems about this project, can i get help from you?

**Comment by KeenNest** [comment]
Hi @Justarrrrr you can text me on LinkedIn -- https://www.linkedin.com/in/ashish-kumar-5a6718172?utm_source=share&utm_campaign=share_via&utm_content=profile&utm_medium=android_app

**Comment by Justarrrrr** [comment]
hi, thank you, but it's regretful that my area cannot support the LinkedIn, can you give me another text way?

**Comment by KeenNest** [comment]
@Justarrrrr what kind of problem you're facing ?

**Comment by Justarrrrr** [comment]
I want use this module, but I don't know if this model is trained or not

**Comment by KeenNest** [comment]
this model is already trained you have to just download it from given link .

**Comment by Justarrrrr** [comment]
and then, how can i use it? i can't write the command and the parameter

**Comment by KeenNest** [comment]
python3 evaluate.py --lr_dir=lr-set --key_dir=key-set --target_dir=hr-set  --output_dir=proj --model_dir=experiments/bix4_keyvsrc_attn --restore_file=pretrained --file_fmt="frame%d.jpg"

**Comment by Justarrrrr** [comment]
Thanks for the info, I would like to know how I can use this model directly without evaluating

**Comment by Justarrrrr** [comment]
hello,KeenNest! Thank you for the help you provided earlier, but now I have a new question. Can you help me answer it? I would like to know why there are three folders in the output of the 'evaluate' model: 'key', 'target', and 'lr'. What is the purpose of these three outputs?

**Comment by KeenNest** [comment]
basically evaluate argument used as a input like lr (low resolution) used as a input from direct live camera and it takes input as a form of frames .  key(high resolution keys) its takes live frame into interval. target used for match the converted frames are match to lr frames ..

**Comment by Justarrrrr** [comment]
I can understand these two inputs, but why we should provide the hr_set, and eventually we get the frames reconstructed  are same resolution as hr_set

**Comment by KeenNest** [comment]
hi @Justarrrrr 
basically, we need  hr_set to check performance of that model.
are u able to produce output from that?

**Comment by Justarrrrr** [comment]
yeah, I can produce the output ,but as my understanding, we just need the low resolution images and some key frames, the hr_set is just needed to compute some metrics like loss rate, is that?

**Comment by KeenNest** [comment]
yes, but can you share some of doubt  i have to produce output files .
and what's you system requirement  you're using .
?

**Comment by Justarrrrr** [comment]
what doubt do you have? I use the Vid4 dataset
--lr_dir=./Vid4/BDx4 --key_dir=./Vid4/GT --target_dir=./Vid4/GT --model_dir=experiments/bix4_keyvsrc_attn --restore_file=pretrained --file_fmt=%08d.png --output_dir=./output
You place the Vid4 dataset in the project folder, and in the end, the output can be obtained in the generated 'output' folder

**Comment by KeenNest** [comment]
i am using python3 evaluate.py --lr_dir=/home/ashish/proj/dataset/lr-set/ --key_dir=/home/ashish/proj/dataset/key-set/ --target_dir=/home/ashish/proj/dataset/hr-set/ --model_dir=experiments/bix4_
keyvsrc_attn/ --restore_file=pretrained --file_fmt="frame%d.png" to run code amd my code got killed after sometime and i am using jetson nano with 4 gb ram.

**Comment by Justarrrrr** [comment]
what's the Traceback?

**Comment by KeenNest** [comment]
ashish@ashish-desktop:~/proj/NeuriCam$ python3 evaluate.py --lr_dir=/home/ashish/proj/dataset/lr-set/ --key_dir=/home/ashish/proj/dataset/key-set/ --target_dir=/home/ashish/proj/dataset/hr-set/ --model_dir=experiments/bix4_keyvsrc_attn/ --restore_file=pretrained --file_fmt="frame%d.png" --output=./output
/usr/local/lib/python3.6/dist-packages/mmcv/__init__.py:21: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  'On January 1, 2023, MMCV will release v2.0.0, in which it will remove '
Creating the dataset...
- done.
load checkpoint from local path: /home/ashish/proj/NeuriCam/model/keyvsrc/spynet_20210409-c6c1bd09.pth
Evaluating keyvsrc
Starting evaluation
Writing results to ./output...
  0%|                                                                                                                                                                                                    | 0/1 [00:00<?, ?it/s]Killed

**Comment by Justarrrrr** [comment]
what dataset you use?

**Comment by KeenNest** [comment]
I created my own dataset ..

**Comment by Justarrrrr** [comment]
May I ask what kind of preprocessing you have applied to your dataset? I tried running the model on the standard Vid4 dataset with some modifications but encountered issues

**Comment by KeenNest** [comment]
first I reduce the size to 160*120, and divide dataset into three parts 
like, lr-set ,hr-set and key-set.
something else i have to do

**Comment by Justarrrrr** [comment]
can you sent your dataset to me have a try?

**Comment by KeenNest** [comment]
https://drive.google.com/drive/folders/1jhmUm9rL8zfY-JJ6Zq3GvagDTzLDsM6-?usp=sharing

what's your machine specification. ?

**Comment by Justarrrrr** [comment]
that's what i want to ask you haha , i use remote machine 2080Ti and 4090, but now i don't have  idle GPU

**Comment by KeenNest** [comment]
I am using. jetson nano: -

128-core NVIDIA Maxwell™ architecture GPU
GPU Max Frequency | 921MHz
CPU | Quad-core ARM® Cortex®-A57 MPCore processor
CPU Max Frequency | 1.43GHz
Memory | 4GB 64-bit LPDDR425.6GB/s

**Comment by Justarrrrr** [comment]
First, you need to create a new folder eg. named "work" under the lr_set and other folders. Secondly, this dataset cannot be processed successfully. I have tried modifying the Vid4 dataset to 160x120, but encountered the same error.

**Comment by KeenNest** [comment]
but i already created the live folder under lr-set. and what the resolution i have to made as for you ,

**Comment by KeenNest** [comment]
Hi @Justarrrrr , any idea why it's not running.

**Comment by Justarrrrr** [comment]
Sorry for the late reply, I haven't found a solution either. I wanted to ask if you have the REDS4 dataset

**Comment by KeenNest** [comment]
no i don't have  .

**Comment by KeenNest** [comment]
hi @Justarrrrr 
can send me the dataset link which you used to test this repo .
and also the command which you're using .

**Comment by Justarrrrr** [comment]
Sure, have you successfully run the model?

**Comment by KeenNest** [comment]
not yet, i done down sampling on that dataset but now getting this error .
RuntimeError: [enforce fail at CPUAllocator.cpp:68] . DefaultCPUAllocator: can't allocate memory: you tried to allocate 5245378560 bytes. Error code 12 (Cannot allocate memory)

**Comment by KeenNest** [comment]
Hi @Justarrrrr 
can you send me the list of library version you install for run this  respo ?
